### PR TITLE
ROX-11315: Bump Scanner to CentOS9

### PR DESCRIPTION
**Pull request by jvdm**
# Description

Bump Scanner build (and test) images to CentOS9, and leave the other images on the same CentOS tag.

The `CENTOS_TAG` file was renamed to `STACKROX_CENTOS_TAG` and the build script and dockerfiles were modified to so that only `stackrox-build` and `rocksdb` image depend on it. All the other images hardcode theirs FROM tags in the Dockerfile themselves. Some refactoring was done in the CircleCI script to emphasize that.

# Tests

None, waiting for the PR to give me the outcome of this.

### CODE REVIEW

**Comment by jvdm** [comment]
Is there a reason why `stackrox-build` uses FROM `rocksdb` and `centos`:

```
FROM quay.io/rhacs-eng/apollo-ci:${ROCKSDB_TAG} as builder

FROM quay.io/centos/centos:${STACKROX_CENTOS_TAG} as base
```

Asking because RocksDB already depends on CentOS.

```
FROM quay.io/centos/centos:${STACKROX_CENTOS_TAG}
```

And the build logic is complex today because of that dependency.

**Review by gavin-stackrox — APPROVED** [review]
LGTM. 

CC @sbostick to incorporate this into his CI migration to github actions work.

**Review by sbostick — APPROVED** [review]
LGTM

Thanks @gavin-stackrox for the heads-up. I'll integrate the changes into into https://github.com/stackrox/rox-ci-image/tree/shane/rs-525-ci-migration.

File: images/scanner-test.Dockerfile [hunk 1]
```diff
@@ -9,40 +9,67 @@ FROM quay.io/rhacs-eng/apollo-ci:${BASE_TAG} as base
...
+ && dnf update -y \
+ && dnf install -y \
         expect \
         gcc \
         gcc-c++ \
-        google-cloud-sdk \
-        google-cloud-sdk-gke-gcloud-auth-plugin \
         jq \
         kubectl \
+        libxcrypt-compat \
```
Comment (RTann) [inline images/scanner-test.Dockerfile:49]: just curious, what is this for?

Comment (RTann) [reply images/scanner-test.Dockerfile:49]: ah ok this is because RHEL 9 is based on Fedora 34

**Review by RTann — APPROVED** [review]

**Pull request merged**